modules/ballerina-core/src/main/java/org/ballerinalang/model/types/BMapType.java

```diff
@@ -33,20 +33,29 @@
  */
 public class BMapType extends BType implements BIndexedType {
 
-    private BType elementType;
+    private BType constraint;
 
     /**
      * Create a type from the given name.
      *
      * @param typeName string name of the type
      */
-    BMapType(String typeName, BType elementType, String pkgPath) {
+    public BMapType(String typeName, BType constraint, String pkgPath) {
         super(typeName, pkgPath, BMap.class);
-        this.elementType = elementType;
+        this.constraint = constraint;
+    }
+
+    public BMapType(BType constraint) {
+        super(TypeConstants.MAP_TNAME, null, BMap.class);
+        this.constraint = constraint;
+    }
+
+    public BType getConstrainedType() {
+        return constraint;
     }
 
     public BType getElementType() {
-        return elementType;
+        return constraint;
```

Comment:
Shall we deprecate this method as we have getConstrainedType() method? we can't remove this now as others using this method.

Reply:
+1